src/Internal.AspNetCore.BuildTools.Tasks/ZipArchive.cs

```diff
@@ -86,8 +87,23 @@ public override bool Execute()
                         return false;
                     }
 
-                    zip.CreateEntryFromFile(file.ItemSpec, entryName);
-                    Log.LogMessage("Added '{0}' to archive", file.ItemSpec);
+                    var entry = zip.CreateEntryFromFile(file.ItemSpec, entryName);
+#if NET46
+#elif NETCOREAPP2_0
+                    if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Yes. Zip has a special flag that indicates which platform created the zip file. (see https://github.com/dotnet/corefx/blob/2f554b13fe67c8cf9b37c70a711de1ba5575638c/src/System.IO.Compression/src/System/IO/Compression/ZipVersion.cs#L16-L29). When this is set to Unix, `unzip` will extract and set file permissions according to the "Extra Attributes" flag. On Windows, we don't have to set these flags because `unzip` will do `if (zipWasCreatedOnWindows) { useDefaultPermissions(); } else { extractPermissionsFromEntry(); }`

Hopefully this nastiness won't be required in the future. See https://github.com/dotnet/corefx/issues/17342